src/renderer/components/node/Node.tsx

```diff
@@ -71,7 +80,10 @@ const NodeInner = memo(({ data, selected }: NodeProps) => {
         [selected, accentColor, regularBorderColor]
     );
 
-    const [validity, setValidity] = useState(VALID);
+    const [validity, setValidity] = useState({
+        isValid: false,
+        reason: 'Validating nodes...',
+    } as Validity);
```

Comment:
I might have mentioned this before already, but it bares repeating: `as` should be a last resort.

```suggestion
    const [validity, setValidity] = useState<Validity>({
        isValid: false,
        reason: 'Validating nodes...',
    });
```